### PR TITLE
Base unit test test start telemetry collection

### PR DESCRIPTION
**Pull request by Twpeak**
**What this PR does / why we need it**:

**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [ ] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go [hunk 1]
```diff
@@ -147,7 +147,8 @@ func Test_getRestConfig(t *testing.T) {
 		{
 			"case 2 use kubepath get config failure",
 			"kubepath",
-			"stat kubepath: no such file or directory",
+			//"stat kubepath: no such file or directory",
```
Comment (kris21he) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go:150]: remove the comment line

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go [hunk 2]
```diff
@@ -186,29 +188,94 @@ func TestNew(t *testing.T) {
...
+		},
+		{
+			"case 5 KubeConfigPath not NULL",
+			&DeviceShifuMetaData{
+				Name:           "test",
+				ConfigFilePath: "etc/edgedevice/config",
+				Namespace:      "test_namespace",
+			},
+			"open /var/run/secrets/kubernetes.io/serviceaccount/token: The system cannot find the path specified.",
+			func() {
```
Comment (kris21he) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go:242]: make a func called intent() if the they are all the same.

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go [hunk 3]
```diff
@@ -186,29 +188,94 @@ func TestNew(t *testing.T) {
...
+				os.Setenv("", "localhost")
+				os.Setenv("KUBERNETES_SERVICE_PORT", "1080")
+				os.Setenv("KUBERNETES_SERVICE_HOST", "127.0.0.1")
+			},
 		},
 	}
 	for _, c := range testCases {
 		t.Run(c.Name, func(t *testing.T) {
+			c.initEnv()
+			defer os.Unsetenv("KUBERNETES_SERVICE_HOST")
```
Comment (kris21he) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go:271]: put these lines into initEnv() function

**Review by tomqin93 — APPROVED** [review]
LGTM

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go [hunk 4]
```diff
@@ -147,7 +147,7 @@ func Test_getRestConfig(t *testing.T) {
 		{
 			"case 2 use kubepath get config failure",
 			"kubepath",
-			"stat kubepath: no such file or directory",
+			"CreateFile kubepath: The system cannot find the file specified.",
```
Comment (tomqin93) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go:150]: let's modify the test to include both error cases

**Review by tomqin93 — COMMENTED** [review]
@Twpeak added 1 comment

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go [hunk 5]
```diff
@@ -221,31 +222,174 @@ func TestNew(t *testing.T) {
...
+	}
+
+	for _, c := range testCases {
+		t.Run(c.Name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			defer func() {
+				if err := recover(); err != nil {
+					assert.Equal(t, c.expErrStr, reflect.ValueOf(err).String())
+				} //内置函数，可以捕捉到函数异常
+			}()
```
Comment (MrLeea-13155bc) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go:381]: Are we really need use recover? 
And. Comment with English

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go [hunk 6]
```diff
@@ -147,15 +148,15 @@ func Test_getRestConfig(t *testing.T) {
...
+			"CreateFile kubepath: The system cannot find the file specified.| stat kubepath: no such file or directory",
 		},
 	}
 
 	for _, c := range testCases {
 		t.Run(c.Name, func(t *testing.T) {
 			config, err := getRestConfig(c.path)
 			if len(c.expErrStr) > 0 {
-				assert.Equal(t, c.expErrStr, err.Error())
+				assert.Equal(t, true, strings.Contains(c.expErrStr, err.Error()))
```
Comment (MrLeea-13155bc) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_config_test.go:159]: how about use assert.Containsf()

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go [hunk 7]
```diff
@@ -221,31 +222,174 @@ func TestNew(t *testing.T) {
...
+		},
+		{
+			"case 6  not NULL AND NewDeviceShifuConfig",
+			&DeviceShifuMetaData{
+				Name:           "test",
+				ConfigFilePath: "etc/edgedevice/config",
+				Namespace:      "test_namespace",
+				KubeConfigPath: "etc/edgedevice/config",
+			},
+			"error loading config file \"etc/edgedevice/config\": read etc/edgedevice/config: The handle is invalid. | error loading config file \"etc/edgedevice/config\": read etc/edgedevice/config: is a directory",
```
Comment (MrLeea-13155bc) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_test.go:248]: How about modify expect err to []string

**Pull request closed without merging**